src/test/optimizer/optimizer_test.cpp

```diff
@@ -120,26 +123,28 @@ TEST_F(OptimizerTest, OptimizesSubqueriesExactlyOnce) {
   // A "rule" that counts how often it was `apply_to()`ed. We use this
   // to check whether SubqueryExpressions pointing to the same LQP before optimization still point to the same (though
   // deep_copy()ed) LQP afterwards
+  size_t counter{0};
   class MockRule : public AbstractRule {
    public:
+    explicit MockRule(size_t& counter) : counter(counter) {}
     std::string name() const override { return "Mock"; }
 
     void apply_to(const std::shared_ptr<AbstractLQPNode>& root) const override { ++counter; }
 
-    mutable size_t counter{0};
+    size_t& counter;
   };
 
-  const auto rule = std::make_shared<MockRule>();
+  auto rule = std::make_unique<MockRule>(counter);
 
   Optimizer optimizer{};
-  optimizer.add_rule(rule);
+  optimizer.add_rule(std::move(rule));
```

Comment:
`add_rule()` uses `std::move()` internally. Is the move here necessary or convention?

Reply:
always gotta move unique_ptrs around, cannot do anything else with those cumbersome simpletons

Reply:
How else would `rule` become an xvalue suitable for being moved?

Reply:
Right, unique_ptrs are used way too rarely.